theano/tensor/basic.py

```diff
@@ -2523,7 +2523,7 @@ def grad(self, inputs, grads):
             new_order = list(x.broadcastable)
             idx = 0
             for i in range(x.ndim):
-                if not new_order[i]:
+                if i not in axis_broadcasted:
```

Comment:
You are right, I forgot a case. This is good, I added an extra tests to this PR for the missing case.